### PR TITLE
Improve docs of tile

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
This is PR for improving docs of functions/links. Related issue: #2182

Modified functions/links:
 - functions/array/tile.py

### CODE REVIEW

File: chainer/functions/array/tile.py [hunk 1]
```diff
@@ -66,13 +66,73 @@ def backward(self, inputs, grads):
...
 
+
     Args:
-        x (chainer.Variable or :class:`numpy.ndarray` or cupy.ndarray):
-            Input data.
-        reps (int or tuple of ints): The number of times for each axis with
-            which x is replicated.
+        x (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variable. If ``x.ndim < d``, ``x`` is treated as
```
Comment (delta2323) [inline chainer/functions/array/tile.py]: The definition of `d` is missing.

File: chainer/functions/array/tile.py [hunk 2]
```diff
@@ -66,13 +66,73 @@ def backward(self, inputs, grads):
...
+            ``d``-dimensional array. For example, when the shape of ``x`` is
+            ``(2,)`` and tiled with 2-dim repetitions, ``x`` is treated as the
+            shape ``(1, 2)``. If ``x.ndim > d``, ``reps`` is treated as
+            ``x.ndim`` by pre-pending 1's.
+        reps (:class:`int` or :class:`tuple` of :class:`int` s):
+            The number of times which x is replicated along each axis.
```
Comment (delta2323) [inline chainer/functions/array/tile.py]: add double quotes to `x`.

File: chainer/functions/array/tile.py [hunk 3]
```diff
@@ -66,13 +66,73 @@ def backward(self, inputs, grads):
...
+
     Args:
-        x (chainer.Variable or :class:`numpy.ndarray` or cupy.ndarray):
-            Input data.
-        reps (int or tuple of ints): The number of times for each axis with
-            which x is replicated.
+        x (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variable. If ``x.ndim < d``, ``x`` is treated as
+            ``d``-dimensional array. For example, when the shape of ``x`` is
```
Comment (delta2323) [inline chainer/functions/array/tile.py]: How does this function treat `x` as a `d`-dim array?

File: chainer/functions/array/tile.py [hunk 4]
```diff
@@ -66,13 +66,73 @@ def backward(self, inputs, grads):
...
+            ``(2,)`` and tiled with 2-dim repetitions, ``x`` is treated as the
+            shape ``(1, 2)``. If ``x.ndim > d``, ``reps`` is treated as
+            ``x.ndim`` by pre-pending 1's.
+        reps (:class:`int` or :class:`tuple` of :class:`int` s):
+            The number of times which x is replicated along each axis.
 
     Returns:
-        ~chainer.Variable: Variable tiled the given array.
+        ~chainer.Variable: The tiled output Variable. Let the length of
+        ``reps`` be ``d``, the output has the dimension of
```
Comment (delta2323) [inline chainer/functions/array/tile.py]: This definition of `d` should be moved forward.

**Comment by keisuke-umezawa** [comment]
@delta2323 
I fixed the code as you reviewd.

File: chainer/functions/array/tile.py [hunk 5]
```diff
@@ -66,13 +66,76 @@ def backward(self, inputs, grads):
...
 
+
     Args:
-        x (chainer.Variable or :class:`numpy.ndarray` or cupy.ndarray):
-            Input data.
-        reps (int or tuple of ints): The number of times for each axis with
-            which x is replicated.
+        x (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variable. Let the length of ``reps`` be ``d`` and
```
Comment (delta2323) [inline chainer/functions/array/tile.py]: I think it is better not to use let for conditioning `x.ndim < d`. So, how about "Let the length of `reps` be `d`. If `x.ndim < d`, " ...?

File: chainer/functions/array/tile.py [hunk 6]
```diff
@@ -66,13 +66,76 @@ def backward(self, inputs, grads):
...
+            ``x.ndim < d``, ``x`` is treated as ``d``-dimensional array by
+            prepending new axes. For example, when the shape of ``x`` is
+            ``(2,)`` and tiled with 2-dim repetitions, ``x`` is treated as the
+            shape ``(1, 2)``. If ``x.ndim > d``, ``reps`` is treated as
+            ``x.ndim`` by pre-pending 1's. For example, when the shape of
```
Comment (delta2323) [inline chainer/functions/array/tile.py]: ... `x.ndim`-dimensional by prepending ...

Comment (delta2323) [inline chainer/functions/array/tile.py]: Or, `N`-dimensional where `N` equals to `x.ndim`.

**Comment by delta2323** [comment]
jenkins, test this please.

File: chainer/functions/array/tile.py (hunk 5 above)
Comment (keisuke-umezawa) [inline chainer/functions/array/tile.py]: Ok! I fixed it.

**Comment by keisuke-umezawa** [comment]
@delta2323 I merged master and fixed the code.

**Pull request merged**